src/stitching/delegateToSchema.ts

```diff
@@ -1,134 +1,158 @@
-    options.fieldName,
-    options.operation,
-    info.fieldNodes,
-    Object.keys(info.fragments).map(
-      fragmentName => info.fragments[fragmentName],
-    ),
-    info.operation.variableDefinitions,
+export function createOperation(
+  targetSchema: GraphQLSchema,
+  targetOperation: 'query' | 'mutation' | 'subscription',
+  roots: Array<OperationRootDefinition>,
+  graphqlContext: { [key: string]: any },
+  documentInfo: GraphQLResolveInfo,
+  transforms?: Array<Transform>,
+): FetcherOperation {
+  const selections: Array<SelectionNode> = roots.map(({ fieldName, info, alias }) => {
+    const newSelections: Array<SelectionNode> = info
+      ? [].concat(...info.fieldNodes.map((field: FieldNode) => field.selectionSet ? field.selectionSet.selections : []))
+      : [];
+
+    const args: Array<ArgumentNode> = info
+      ? [].concat( ...info.fieldNodes.map((field: FieldNode) => field.arguments || []))
+      : [];
+
+    const rootSelectionSet = newSelections.length > 0
+      ? {
+        kind: Kind.SELECTION_SET,
+        selections: newSelections
+      }
+      : null;
+
+    const rootField: FieldNode = {
+      kind: Kind.FIELD,
+      name: {
+        kind: Kind.NAME,
+        value: fieldName,
+      },
+      alias: alias
+       ? {
+         kind: Kind.NAME,
+         value: alias
+       }
+       : null,
+       selectionSet: rootSelectionSet,
+       arguments: args
+    };
+
+    return rootField;
+  }, []);
+
+  const selectionSet: SelectionSetNode = {
+    kind: Kind.SELECTION_SET,
+    selections,
+  };
+
+  const operationDefinition: OperationDefinitionNode = {
+    kind: Kind.OPERATION_DEFINITION,
+    operation: targetOperation,
+    variableDefinitions: documentInfo.operation.variableDefinitions,
+    selectionSet,
+  };
+
+  const fragments = Object.keys(documentInfo.fragments).map(
+    fragmentName => documentInfo.fragments[fragmentName],
   );
 
+  const document = {
+    kind: Kind.DOCUMENT,
+    definitions: [operationDefinition, ...fragments],
+  };
+
   const rawRequest: Request = {
-    document: rawDocument,
-    variables: info.variableValues as Record<string, any>,
+    document,
+    variables: documentInfo.variableValues as Record<string, any>,
   };
 
-  const transforms = [
-    ...(options.transforms || []),
-    AddArgumentsAsVariables(options.schema, args),
-    FilterToSchema(options.schema),
-    AddTypenameToAbstract(options.schema),
-    CheckResultAndHandleErrors(info, options.fieldName),
+  transforms = [
+    ...(transforms || []),
+    AddArgumentsAsVariables(targetSchema, roots),
+    FilterToSchema(targetSchema),
+    AddTypenameToAbstract(targetSchema)
   ];
 
-  const processedRequest = applyRequestTransforms(rawRequest, transforms);
+  const { document: query, variables } = applyRequestTransforms(rawRequest, transforms);
 
-  const errors = validate(options.schema, processedRequest.document);
+  return {
+    query,
+    variables,
+    context: graphqlContext,
```

Comment:
Should we return `context: graphqlContext` or `context: { graphqlContext }`? My original intention was the second because then you can take the return object and pass it directly to a fetcher.

Reply:
Or should we remote context all together, and just have this function return `{ query, variables` or `{ document, variables }`?

Reply:
Just document and variables I think.